src/promptflow-tools/promptflow/tools/aoai.py

```diff
@@ -78,14 +87,13 @@ def completion(
             # Logit bias must be a dict if we passed it to openai api.
             logit_bias=logit_bias if logit_bias else {},
             user=user,
-            headers={"ms-azure-ai-promptflow-called-from": "aoai-tool"},
-            **self._connection_dict,
-        )
+            extra_headers={"ms-azure-ai-promptflow-called-from": "aoai-tool"})
+
         if stream:
             def generator():
                 for chunk in response:
                     if chunk.choices:
-                        yield getattr(chunk.choices[0], "text", "")
+                        yield chunk.choices[0].text or ""
```

Comment:
in order to keep the same logic as before I think you may need to consider the case that `chunk.choices[0]` may not have `text` property.

Reply:
refined.
`
yield chunk.choices[0].text if hasattr(chunk.choices[0], 'text') and chunk.choices[0].text is not None else ""
`